Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,103 @@
+# Kubernetes On-Premises Deployment From Scratch -- Start to Finish #
+This guide will walk you through deploying *Kubernetes 1.8* on a Linux master and join two Windows nodes to it without a cloud provider.
+
+**Note**: The guide will assume a local working directory of `~/kube` for the Kubernetes setup. If you choose a different directory, just replace any references to that path.
+
+## Building the Binaries ##
+We will begin with a Linux master node and a Windows worker node on RS3+, both of which are assumed to be VMs.
+
+First, install all of the pre-requisites on the master:
+
+    # apt-get install git build-essentials docker.io
+
+### Building Kubernetes ###
+We will need to build the `kubelet` and `kubeproxy` binaries for Windows from scratch _cross-compiling from Linux_, so let's [set up a Go environment](https://golang.org/doc/install#install) for that, first. This guide assumes Go 1.9 and has not been tested with anything else. With your Go environment ready (don't forget to set your `$GOPATH`!), we can build the Windows binaries:
+
+```bash
+$ K8SREPO="github.com/madhanrm/kubernetes"
+$ go get -d $K8SREPO
+$ cd $GOPATH/src/$K8SREPO
+$ KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kubectl
+$ cp ~/_output/bin/kubectl.exe ~/kube/
+$ git checkout cniwindows
+$ make clean && KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kubelet
+$ cp ~/_output/bin/kubelet.exe ~/kube/
+$ git checkout winkernelproxy 
+$ KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kube-proxy
+$ cp ~/_output/bin/kube-proxy.exe ~/kube/
+```
+
+_You may run into problems with various "permission denied" errors when building these binaries. This is a bug in the Kubernetes build process for Windows, which uses binaries that exist, but are size-zero, and thus don't have executable permissions on them. The solution, unfortunately, is to run `chmod +x _output/bin/*` after every time (~4 times) you hit the error._

```

Comment:
"You may run into problems" - I did :) I think this is going to happen for most people, so this warning/guidance should come before the build steps so that the reader is already expecting the make commands to fail and knows what to do when they do fail.

Reply:
Also, don't ask me why...but my binary ended up being output to the following location:
_output/local/bin/windows/amd64/kubectl.exe